src/components/AppProvider/utilities/withAppProvider/withAppProvider.tsx

```diff
@@ -1,94 +1,113 @@
 import * as React from 'react';
 import hoistStatics from 'hoist-non-react-statics';
 import {ClientApplication} from '@shopify/app-bridge';
-import merge from '../../../../utilities/merge';
 import Intl from '../Intl';
 import Link from '../Link';
-import {StickyManager} from '../withSticky';
+import StickyManager from '../StickyManager';
 import ScrollLockManager from '../ScrollLockManager';
 import {
   ThemeContext,
   Consumer as ThemeProviderConsumer,
 } from '../../../ThemeProvider';
-import {polarisAppProviderContextTypes} from '../../types';
+import AppProviderContext from '../../Context';
+
+export type ReactComponent<P, C> =
+  | React.ComponentClass<P> & C
+  | React.SFC<P> & C;
 
 export interface WithAppProviderProps {
   polaris: {
     intl: Intl;
     link: Link;
     stickyManager: StickyManager;
     scrollLockManager: ScrollLockManager;
-    theme: ThemeContext;
-    appBridge: ClientApplication<{}>;
-    subscribe(callback: () => void): void;
-    unsubscribe(callback: () => void): void;
+    theme?: ThemeContext;
+    appBridge?: ClientApplication<{}>;
   };
 }
 
-export default function withAppProvider<OwnProps>() {
-  return function addProvider<C>(
-    WrappedComponent:
-      | React.ComponentClass<OwnProps & WithAppProviderProps> & C
-      | React.SFC<OwnProps & WithAppProviderProps> & C,
-  ): React.ComponentClass<OwnProps> & C {
-    class WithProvider extends React.Component<OwnProps, never> {
-      static contextTypes = WrappedComponent.contextTypes
-        ? merge(WrappedComponent.contextTypes, polarisAppProviderContextTypes)
-        : polarisAppProviderContextTypes;
+export interface Options {
+  inScrollable?: boolean;
+}
 
-      componentDidMount() {
-        const {
-          polaris: {subscribe: subscribeToPolaris},
-        } = this.context;
+function withScrollable<P, T>(WrappedComponent: ReactComponent<P, T>) {
```

Comment:
small `HoC` to provide a new `StickyManager` to scrollable

Reply:
Why is this necessary?

Reply:
This was previously it's own HoC that duplicated some logic from `withAppProvider` and added some additional logic. `scrollable` requires it's own `stickyManger` and this function creates that and provides it all children consuming the HoC